[tensorexpr] Enabled aten::stack in the fuser pass with static shapes

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74077
* #73801

Differential Revision: [D34808051](https://our.internmc.facebook.com/intern/diff/D34808051)

huiguoo committed:
[tensorexpr] Enabled aten::stack in the fuser pass with static shapes

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34808051).

navahgar commented on test/test_jit_fuser_te.py:
```diff
@@ -737,6 +737,22 @@ def foo(hx, cx):
                 # FileCheck().check("FusedConcat").check_next("return").run(str(graph))
 
+    def test_stack(self):
+        # "aten::stack fusion is not enabled yet with dynamic shapes"
```
What is the reason for this restriction?

huiguoo replied:
The shape computation formula for aten::stack in JIT is not added in this PR. Will do it in the next PR. It seems a bit compound to have both changes for static and dynamic shapes in one PR.

huiguoo committed:
Update on "[tensorexpr] Enabled aten::stack in the fuser pass with static shapes"


Differential Revision: [D34808051](https://our.internmc.facebook.com/intern/diff/D34808051)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34808051).

navahgar reviewed (approved):
LGTM

huiguoo committed:
Update on "[tensorexpr] Enabled aten::stack in the fuser pass with static shapes"


Differential Revision: [D34808051](https://our.internmc.facebook.com/intern/diff/D34808051)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34808051).

huiguoo committed:
Update on "[tensorexpr] Enabled aten::stack in the fuser pass with static shapes"


Differential Revision: [D34808051](https://our.internmc.facebook.com/intern/diff/D34808051)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34808051).

huiguoo committed:
Update on "[tensorexpr] Enabled aten::stack in the fuser pass with static shapes"


Differential Revision: [D34808051](https://our.internmc.facebook.com/intern/diff/D34808051)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34808051).

huiguoo committed:
Update on "[tensorexpr] Enabled aten::stack in the fuser pass with static shapes"


Differential Revision: [D34808051](https://our.internmc.facebook.com/intern/diff/D34808051)

[ghstack-poisoned]

huiguoo commented:
@huiguoo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34808051).
